include/lsst/meas/astrom/pessimisticPatternMatcherUtils.h

```diff
@@ -0,0 +1,69 @@
+// -*- LSST-C++ -*-
+
+/*
+ * This file is part of meas_astrom.
+ *
+ * Developed for the LSST Data Management System.
+ * This product includes software developed by the LSST Project
+ * (https://www.lsst.org).
+ * See the COPYRIGHT file at the top-level directory of this distribution
+ * for details of code ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <ndarray.h>
+
+namespace lsst {
+namespace meas {
+namespace astrom {
+
+/**
+ * Check the opening angle between the first spoke of our pattern for the
+ * source object against the reference object.
+ *
+ * This method makes heavy use of the small angle approximation to perform
+ * the comparison.
+ *
+ * @param[in] cos_theta_src Cosine of the angle between the current candidate
+ *     source spoke and the first spoke.
+ * @param[in] sin_theta_src Sine of the angle between the current candidate
+ *     source spoke and the first spoke.
+ * @param[in] ref_ctr 3 vector of the candidate reference center
+ * @param[in] ref_ctr_id id lookup of the ref_ctr into the master reference
+ *     array.
+ * @param[in] proj_ref_ctr_delta Plane projected first spoke in the reference
+ *     pattern using the pattern center as normal.
+ * @param[in] proj_ref_ctr_dist_sq Squared length of the projected vector.
+ * @param[in] ref_dist_idx_array Indices sorted by the delta distance between
+ *     the source spoke we are trying to test and the candidate reference
+ *     spokes.
+ * @param[in] ref_id_array Array of id lookups into the master reference array
+ *     that our center id object is paired with.
```

Comment:
Please add a note here about why this is uint16 (as you said, the higher level code trims the arrays due to memory limits): it's generally expected that indexes are size_t.